# Investigating Replicate with Irreproducible Discovery Rate (IDR):  
As with any expriment it is always a good idea to have multiple measurements of any object of interest. If we are measuring any genuine effect in  
some experiment, we should get similar observation if we have to repeat the experiment. similary, if we observing a peak in a particular genomic region in a chip-seq experiment  
we should get almost equivalent measure of peak for that region in a replicate experiment. And ideally, this observation of peak mapping in replicate  
samples should extend to all genuine peaks. This way we are sure of peak signal as a true peak. IDR serve as a metric to identify genuine peaks based  
on their reproducibility across replicates. It is a statistical framework and the main idea behind this is to generate a `peak` list of each replicate  
with lenient threshold. Doing this will generate true genuine `peaks` as well as some noise `peaks`. If we rank `peaks` from all the replicate and look  
for the correlation, we observe that the gennuine `peaks` are all well correlated while noise `peaks` are not.  

## we will look at the consistency between replicates:  
fist step is to rerun `macs2 call` with relaxed p-value to identify lot many peaks; default is 1e-5, but we take 1e-3  
```bash
mkdir -p chipseq-project/IDR/peaks
control=NRF1_INPUT_TKO
for sample in NRF1_CHIP_TKO_1 NRF1_CHIP_TKO_2   
	do
	macs2 callpeak -t reads/${sample}.bam\
	-c reads/${control}.bam \
	-f BAM \
	-g mm \
	 --outdir chipseq-project/IDR/peaks/ \
	 --bdg \
	 -p 1e-3 \
	 -n ${sample}  2> chipseq-project/IDR/logs/${sample}_macs2_report.log
	done
  ```
  Rerun `macs2 callpeak` for `WT` sample
  ```bash
  control=NRF1_INPUT_WT
  for sample in NRF1_CHIP_WT_1 NRF1_CHIP_WT_2  
	do
	macs2 callpeak -t reads/${sample}.bam\
	-c reads/${control}.bam \
	-f BAM \
	-g mm \
	 --outdir chipseq-project/IDR/peaks/ \
	 --bdg \
	 -p 1e-3 \
	 -n ${sample}  2> chipseq-project/IDR/logs/${sample}_macs2_report.log
	done
	```
	
	
Next, we sort ".narrowPeak" files according to the "-log(10) or 8th colum".  
```bash
for sample in NRF1_CHIP_WT_1 NRF1_CHIP_WT_2 NRF1_CHIP_TKO_1 NRF1_CHIP_TKO_2
	do
	sort -k8,8nr chipseq-project/IDR/peaks/${sample}_peaks.narrowPeak > chipseq-project/IDR/${sample}_sorted_peaks.narrowPeak
	done
```  

 Now that we have sorted our ".narrowPeak" files according to the p-values, we are ready to run `idr`.  
 ```bash
 cd chipseq-project/IDR
 idr --samples NRF1_CHIP_WT_1_sorted_peaks.narrowPeak NRF1_CHIP_WT_2_sorted_peaks.narrowPeak \
--input-file-type narrowPeak \
--rank p.value \
--output-file NRF1_WT_idr \
--plot \
--log-output-file NRF1_WT.idr.log
```
Similarly, for `TKO` sample  

```bash
idr --samples NRF1_CHIP_TKO_1_sorted_peaks.narrowPeak NRF1_CHIP_TKO_2_sorted_peaks.narrowPeak \
--input-file-type narrowPeak \
--rank p.value \
--output-file NRF1_TKO_idr \
--plot \
--log-output-file NRF1_TKO.idr.log
```

We can look at the `log` files and determine how many shared regions have passed the IDR -cutoff of 0.05.  
Also, we can look at the out put image files from `idr` to assess the reproducibility between samples.  

![NRF1_WT_idr](https://user-images.githubusercontent.com/85447250/169447760-c869a6ed-3e0e-47f2-8769-fff826f75b5d.png)
![NRF1_TKO_idr](https://user-images.githubusercontent.com/85447250/169447771-e4413711-4d3c-419e-a105-5eebac0d6c84.png)

  
  
