## Creating bigWig files  
Now that we have filtered, aligned, and assessed the Chip-seq specific quality metrices, we are ready to look at the graphic representation of peaks  
regions. For this first we need to create BigWig files. Here, we have created bigWig files for each sample replicate normalised to the `Input`.  
Although, we can create bigWig files of individual samples without normalising them to their respective `Input` files. We can use two methods from  
`deepTools` suite 1. bamCompare 2. bamCoverage. Here we will use `bamCompare` method.  

```bash
mkdir -p chipseq-project/visualization/{logs,bigwig}
control=NRF1_INPUT_TKO
for sample in NRF1_CHIP_TKO_1 NRF1_CHIP_TKO_2
	do
	bamCompare -b1 chipseq-project/data/bams/${sample}.bam \
	-b2 chipseq-project/data/bams/${control}.bam \
	-o chipseq-project/visualisation/bigwig/${sample}_bwNorm.bw \
	--binSize 20 \
	--normalizeUsing BPM \
	--smoothLength 60 \
	--extendReads 150 \
	--centerReads \
	--scaleFactorsMethod None \
	-p 6 2> chipseq-project/visualisation/logs/${sample}_bamCompare.log
	done
  ```
  
  Similarly, We can repeat the `bamCompare` for `WT` samples.  
  ```bash
  control=NRF1_INPUT_WT
for sample in NRF1_CHIP_WT_1 NRF1_CHIP_WT_2
	do
	bamCompare -b1 chipseq-project/data/bams/${sample}.bam \
	-b2 chipseq-project/data/bams/${control}.bam \
	-o chipseq-project/visualisation/bigwig/${sample}_bwNorm.bw \
	--binSize 20 \
	--normalizeUsing BPM \
	--smoothLength 60 \
	--extendReads 150 \
	--centerReads \
	--scaleFactorsMethod None \
	-p 6 2> chipseq-project/visualisation/logs/${sample}_bamCompare.log
	done
  ```
once we have created the bigWig files we do proceed with the visualisation step.We need to run `computeMatrix reference-point` method from `bedTools`  
to compute the matrix for `TKO` and `WT` sample replicates.  
```bash
computeMatrix reference-point \
-S chipseq-project/visualisation/bigwig/NRF1_CHIP_TKO_1_bwNorm.bw chipseq-project/visualisation/bigwig/NRF1_CHIP_TKO_2_bwNorm.bw \
-R chipseq-project/visualisation/mm10_gene.bed \
-b 1000 \
-a 1000 \
--skipZeros \
-o chipseq-project/visualisation/matrix_NRF1_TKO_TSS.gz \
-p 6 \
--outFileSortedRegions chipseq-project/visualisation/NRF1_TKO_regions_TSS.bed
```
Repeat `computeMatrix reference-point` method for `WT` samples.  
```bash
computeMatrix reference-point \
-S chipseq-project/visualisation/bigwig/NRF1_CHIP_WT_1_bwNorm.bw chipseq-project/visualisation/bigwig/NRF1_CHIP_WT_2_bwNorm.bw \
-R chipseq-project/visualisation/mm10_gene.bed \
-b 1000 \
-a 1000 \
--skipZeros \
-o chipseq-project/visualisation/matrix_NRF1_WT_TSS.gz \
-p 6 \
--outFileSortedRegions chipseq-project/visualisation/NRF1_WT_regions_TSS.bed
```

Once `computeMatrix reference-point` is done for both `TKO` and `WT`, we can proceed towards profile plotting.   
```bash
plotProfile -m chipseq-project/visualisation/matrix_NRF1_TKO_TSS.gz \
-out chipseq-project/visualisation/TSS_TKO_profile.png \
--perGroup \
--colors green purple \
--plotTitle "" --samplesLabel "Rep1" "Rep2" \
--refPointLabel "TSS" \
-T "NRF1_TKO read density" \
-z ""
```
![TSS_TKO_profile](https://user-images.githubusercontent.com/85447250/169661816-0724f51f-b99f-406f-87d4-750a1c6d7294.png)

```bash
plotProfile -m chipseq-project/visualisation/matrix_NRF1_WT_TSS.gz \
-out chipseq-project/visualisation/TSS_WT_profile.png \
--perGroup \
--colors green purple \
--plotTitle "" --samplesLabel "Rep1" "Rep2" \
--refPointLabel "WT" \
-T "NRF1_WT read density" \
-z ""
```
![TSS_WT_profile](https://user-images.githubusercontent.com/85447250/169661847-1bacf82e-3fb1-466b-a0b7-537cad333d65.png)

Similarly, we can plot heatmaps of binding TF around TSS (it is just another way of depicting "plotProfile" output).  
```bash
plotHeatmap -m chipseq-project/visualisation/matrix_NRF1_TKO_TSS.gz \
-out chipseq-project/visualisation/TSS_TKO_heatmap.png \
--colorMap RdBu \
--whatToShow 'heatmap and colorbar' \
--zMin -4 --zMax 4
```
![TSS_TKO_heatmap](https://user-images.githubusercontent.com/85447250/169661965-da026f1d-9462-47d5-b389-9f4955c856f8.png)


  
